pkg/app/app.go

```diff
@@ -193,23 +217,43 @@ func (app *App) setupRepo() (bool, error) {
 		}
 
 		if !shouldInitRepo {
-			// check if we have a recent repo we can open
-			for _, repoDir := range app.Config.GetAppState().RecentRepos {
-				if isRepo, _ := isDirectoryAGitRepository(repoDir); isRepo {
-					if err := os.Chdir(repoDir); err == nil {
-						return true, nil
-					}
-				}
+			// Attempt to open a recent repo, exit if no repo could be opened
+			if didOpenRepo := openRecentRepo(app); !didOpenRepo {
+				fmt.Println(app.Tr.NoRecentRepositories)
+				os.Exit(1)
 			}
 
-			fmt.Println(app.Tr.NoRecentRepositories)
-			os.Exit(1)
+			return true, nil
 		}
 		if err := app.OSCommand.Cmd.New("git init " + initialBranch).Run(); err != nil {
 			return false, err
 		}
 	}
 
+	// Run this afterward so that the previous repo creation steps can run without this interfering
+	if isBare, err := isBareRepo(app.OSCommand); isBare {
+		if err != nil {
+			return false, err
+		}
+
+		fmt.Print(app.Tr.BareRepo)
+
+		response, _ := bufio.NewReader(os.Stdin).ReadString('\n')
+		shouldOpenRecent := strings.Trim(response, " \r\n") == "y"
+
+		if shouldOpenRecent {
+			if didOpenRepo := openRecentRepo(app); !didOpenRepo {
+				fmt.Println(app.Tr.NoRecentRepositories)
+				os.Exit(1)
+			}
+		
+			// We managed to open a recent repo, continue as usual
+			return true, nil
+		} else {
```

Comment:
Unnecessary `else` branch, the `if` branch will surely `return` (or exit).

Reply:
I don't mean to say to remove the `os.Exit(0)` below, just that it shouldn't be in the `else` branch, rather after the `if`.

Reply:
Added it for readability, seeing as theres a lot of branching logic going on here. Can always remove if you really want.

Reply:
From a readability standpoint I'd rather have the inverse, then

```
if !shouldOpenRecent {
    os.Exit(0)
}

if didOpenRepo := openRecentRepo(app); didOpenRepo {
    return true, nil
}

fmt.Println(app.Tr.NoRecentRepositories)
os.Exit(1)
```

Then we don't need the comment explaining why are we returning `true, nil`, it's clear from the condition